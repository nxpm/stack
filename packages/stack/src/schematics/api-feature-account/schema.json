{
  "$schema": "http://json-schema.org/draft-07/schema",
  "id": "ApiFeatureUser",
  "title": "",
  "type": "object",
  "properties": {
    "appName": {
      "type": "string",
      "description": "Name of the app to connect to",
      "alias": "a"
    },
    "directory": {
      "type": "string",
      "description": "A directory where the project is placed",
      "alias": "d"
    }
  },
  "required": []
}
