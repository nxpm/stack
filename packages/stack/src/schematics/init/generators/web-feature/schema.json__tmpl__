{
  "cli": "nx",
  "id": "web-feature",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "Library name",
      "$default": {
        "$source": "argv",
        "index": 0
      }
    },
    "appName": {
      "type": "string",
      "description": "Application name",
      "default": "web"
    }
  },
  "required": ["name", "appName"]
}
