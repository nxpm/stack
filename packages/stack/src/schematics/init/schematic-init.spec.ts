import { Tree } from '@angular-devkit/schematics'
import { SchematicTestRunner } from '@angular-devkit/schematics/testing'
import { createEmptyWorkspace } from '@nrwl/workspace/testing'
import { join } from 'path'

import { InitSchematicSchema } from './schema'

describe('init schematic', () => {
  let appTree: Tree
  const options: InitSchematicSchema = { name: 'test' }

  const testRunner = new SchematicTestRunner('@nxpm/init', join(__dirname, '../../../collection.json'))

  beforeEach(() => {
    appTree = createEmptyWorkspace(Tree.empty())
  })

  it('should run successfully', async () => {
    await expect(testRunner.runSchematicAsync('init', options, appTree).toPromise()).resolves.not.toThrowError()
  })
})
