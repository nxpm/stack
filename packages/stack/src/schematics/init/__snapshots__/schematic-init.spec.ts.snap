// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`init schematic should run successfully 1`] = `
"{
  \\"npmScope\\": \\"proj\\",
  \\"projects\\": {
    \\"api\\": {
      \\"tags\\": []
    },
    \\"api-e2e\\": {
      \\"tags\\": []
    },
    \\"api-auth-data-access\\": {
      \\"tags\\": [\\"scope:api\\", \\"type:data-access\\"]
    },
    \\"api-auth-feature\\": {
      \\"tags\\": [\\"scope:api\\", \\"type:feature\\"]
    },
    \\"api-core-data-access\\": {
      \\"tags\\": [\\"scope:api\\", \\"type:data-access\\"]
    },
    \\"api-core-feature\\": {
      \\"tags\\": [\\"scope:api\\", \\"type:feature\\"]
    },
    \\"test\\": {
      \\"tags\\": []
    },
    \\"test-e2e\\": {
      \\"tags\\": [],
      \\"implicitDependencies\\": [\\"test\\"]
    },
    \\"test-assets\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:assets\\"]
    },
    \\"test-auth-data-access\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:data-access\\"]
    },
    \\"test-auth-ui\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:ui\\"]
    },
    \\"test-core-data-access\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:data-access\\"]
    },
    \\"test-about-feature\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:feature\\"]
    },
    \\"test-auth-feature\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:feature\\"]
    },
    \\"test-core-feature\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:feature\\"]
    },
    \\"test-dashboard-feature\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:feature\\"]
    },
    \\"test-shell-feature\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:feature\\"]
    },
    \\"test-layout\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:layout\\"]
    },
    \\"test-style\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:style\\"]
    },
    \\"test-ui-form\\": {
      \\"tags\\": [\\"scope:test\\", \\"type:ui\\"]
    }
  },
  \\"affected\\": {
    \\"defaultBase\\": \\"main\\"
  },
  \\"tasksRunnerOptions\\": {
    \\"default\\": {
      \\"runner\\": \\"@nrwl/workspace/tasks-runners/default\\",
      \\"options\\": {
        \\"cacheableOperations\\": [\\"build\\", \\"lint\\", \\"test\\", \\"e2e\\"]
      }
    }
  },
  \\"plugins\\": {
    \\"@nxpm/stack\\": {
      \\"api\\": {
        \\"project\\": \\"api\\"
      },
      \\"web\\": {
        \\"project\\": \\"test\\"
      }
    }
  }
}
"
`;

exports[`init schematic should run successfully 2`] = `
"{
  \\"version\\": 1,
  \\"projects\\": {
    \\"api\\": {
      \\"root\\": \\"apps/api\\",
      \\"sourceRoot\\": \\"apps/api/src\\",
      \\"projectType\\": \\"application\\",
      \\"prefix\\": \\"api\\",
      \\"schematics\\": {},
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@nrwl/node:build\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/apps/api\\",
            \\"main\\": \\"apps/api/src/main.ts\\",
            \\"tsConfig\\": \\"apps/api/tsconfig.app.json\\",
            \\"assets\\": [\\"apps/api/src/assets\\"]
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"optimization\\": true,
              \\"extractLicenses\\": true,
              \\"inspect\\": false,
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"apps/api/src/environments/environment.ts\\",
                  \\"with\\": \\"apps/api/src/environments/environment.prod.ts\\"
                }
              ]
            }
          }
        },
        \\"serve\\": {
          \\"builder\\": \\"@nrwl/node:execute\\",
          \\"options\\": {
            \\"buildTarget\\": \\"api:build\\"
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"apps/api/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"apps/api/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      }
    },
    \\"api-e2e\\": {
      \\"root\\": \\"apps/api-e2e\\",
      \\"sourceRoot\\": \\"apps/api-e2e/src\\",
      \\"projectType\\": \\"application\\",
      \\"prefix\\": \\"api-e2e\\",
      \\"schematics\\": {},
      \\"architect\\": {
        \\"e2e\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"apps/api-e2e/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:lint\\",
          \\"options\\": {
            \\"linter\\": \\"eslint\\",
            \\"tsConfig\\": [\\"apps/api-e2e/tsconfig.spec.json\\"],
            \\"exclude\\": [\\"**/node_modules/**\\", \\"!apps/api-e2e/**/*\\"]
          }
        }
      }
    },
    \\"api-auth-data-access\\": {
      \\"root\\": \\"libs/api/auth/data-access\\",
      \\"sourceRoot\\": \\"libs/api/auth/data-access/src\\",
      \\"projectType\\": \\"library\\",
      \\"schematics\\": {},
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/api/auth/data-access/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/api/auth/data-access/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      }
    },
    \\"api-auth-feature\\": {
      \\"root\\": \\"libs/api/auth/feature\\",
      \\"sourceRoot\\": \\"libs/api/auth/feature/src\\",
      \\"projectType\\": \\"library\\",
      \\"schematics\\": {},
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/api/auth/feature/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/api/auth/feature/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      }
    },
    \\"api-core-data-access\\": {
      \\"root\\": \\"libs/api/core/data-access\\",
      \\"sourceRoot\\": \\"libs/api/core/data-access/src\\",
      \\"projectType\\": \\"library\\",
      \\"schematics\\": {},
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/api/core/data-access/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/api/core/data-access/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      }
    },
    \\"api-core-feature\\": {
      \\"root\\": \\"libs/api/core/feature\\",
      \\"sourceRoot\\": \\"libs/api/core/feature/src\\",
      \\"projectType\\": \\"library\\",
      \\"schematics\\": {},
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/api/core/feature/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/api/core/feature/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      }
    },
    \\"test\\": {
      \\"projectType\\": \\"application\\",
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      },
      \\"root\\": \\"apps/test\\",
      \\"sourceRoot\\": \\"apps/test/src\\",
      \\"prefix\\": \\"proj\\",
      \\"architect\\": {
        \\"build\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:browser\\",
          \\"options\\": {
            \\"outputPath\\": \\"dist/apps/test\\",
            \\"index\\": \\"apps/test/src/index.html\\",
            \\"main\\": \\"apps/test/src/main.ts\\",
            \\"polyfills\\": \\"apps/test/src/polyfills.ts\\",
            \\"tsConfig\\": \\"apps/test/tsconfig.app.json\\",
            \\"aot\\": true,
            \\"assets\\": [
              {
                \\"glob\\": \\"favicon.ico\\",
                \\"input\\": \\"libs/test/assets/src\\",
                \\"output\\": \\"./\\"
              },
              {
                \\"glob\\": \\"**/*\\",
                \\"input\\": \\"libs/test/assets/src/assets\\",
                \\"output\\": \\"assets\\"
              }
            ],
            \\"styles\\": [\\"libs/test/style/src/index.scss\\"],
            \\"scripts\\": [],
            \\"allowedCommonJsDependencies\\": [\\"graphql-tag\\", \\"subscriptions-transport-ws\\", \\"zen-observable\\"]
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"fileReplacements\\": [
                {
                  \\"replace\\": \\"libs/test/core/feature/src/environments/environment.ts\\",
                  \\"with\\": \\"libs/test/core/feature/src/environments/environment.prod.ts\\"
                }
              ],
              \\"optimization\\": true,
              \\"outputHashing\\": \\"all\\",
              \\"sourceMap\\": false,
              \\"extractCss\\": true,
              \\"namedChunks\\": false,
              \\"extractLicenses\\": true,
              \\"vendorChunk\\": false,
              \\"buildOptimizer\\": true,
              \\"budgets\\": [
                {
                  \\"type\\": \\"initial\\",
                  \\"maximumWarning\\": \\"2mb\\",
                  \\"maximumError\\": \\"5mb\\"
                },
                {
                  \\"type\\": \\"anyComponentStyle\\",
                  \\"maximumWarning\\": \\"6kb\\",
                  \\"maximumError\\": \\"10kb\\"
                }
              ]
            }
          }
        },
        \\"serve\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:dev-server\\",
          \\"options\\": {
            \\"browserTarget\\": \\"test:build\\",
            \\"proxyConfig\\": \\"apps/test/proxy.conf.js\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"browserTarget\\": \\"test:build:production\\"
            }
          }
        },
        \\"extract-i18n\\": {
          \\"builder\\": \\"@angular-devkit/build-angular:extract-i18n\\",
          \\"options\\": {
            \\"browserTarget\\": \\"test:build\\"
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"apps/test/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"apps/test/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      }
    },
    \\"test-e2e\\": {
      \\"root\\": \\"apps/test-e2e\\",
      \\"sourceRoot\\": \\"apps/test-e2e/src\\",
      \\"projectType\\": \\"application\\",
      \\"architect\\": {
        \\"e2e\\": {
          \\"builder\\": \\"@nrwl/cypress:cypress\\",
          \\"options\\": {
            \\"cypressConfig\\": \\"apps/test-e2e/cypress.json\\",
            \\"tsConfig\\": \\"apps/test-e2e/tsconfig.e2e.json\\",
            \\"devServerTarget\\": \\"test:serve\\"
          },
          \\"configurations\\": {
            \\"production\\": {
              \\"devServerTarget\\": \\"test:serve:production\\"
            }
          }
        },
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"apps/test-e2e/**/*.{js,ts}\\"]
          }
        }
      }
    },
    \\"test-assets\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/assets\\",
      \\"sourceRoot\\": \\"libs/test/assets/src\\",
      \\"prefix\\": \\"assets\\",
      \\"architect\\": {},
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-auth-data-access\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/auth/data-access\\",
      \\"sourceRoot\\": \\"libs/test/auth/data-access/src\\",
      \\"prefix\\": \\"auth\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/auth/data-access/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/auth/data-access/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-auth-ui\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/auth/ui\\",
      \\"sourceRoot\\": \\"libs/test/auth/ui/src\\",
      \\"prefix\\": \\"auth\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/auth/ui/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/auth/ui/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-core-data-access\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/core/data-access\\",
      \\"sourceRoot\\": \\"libs/test/core/data-access/src\\",
      \\"prefix\\": \\"core\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/core/data-access/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/core/data-access/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-about-feature\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/about/feature\\",
      \\"sourceRoot\\": \\"libs/test/about/feature/src\\",
      \\"prefix\\": \\"about\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/about/feature/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/about/feature/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-auth-feature\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/auth/feature\\",
      \\"sourceRoot\\": \\"libs/test/auth/feature/src\\",
      \\"prefix\\": \\"auth\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/auth/feature/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/auth/feature/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-core-feature\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/core/feature\\",
      \\"sourceRoot\\": \\"libs/test/core/feature/src\\",
      \\"prefix\\": \\"core\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/core/feature/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/core/feature/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-dashboard-feature\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/dashboard/feature\\",
      \\"sourceRoot\\": \\"libs/test/dashboard/feature/src\\",
      \\"prefix\\": \\"dashboard\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/dashboard/feature/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/dashboard/feature/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-shell-feature\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/shell/feature\\",
      \\"sourceRoot\\": \\"libs/test/shell/feature/src\\",
      \\"prefix\\": \\"shell\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/shell/feature/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/shell/feature/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-layout\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/layout\\",
      \\"sourceRoot\\": \\"libs/test/layout/src\\",
      \\"prefix\\": \\"layout\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/layout/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/layout/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-style\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/style\\",
      \\"sourceRoot\\": \\"libs/test/style/src\\",
      \\"prefix\\": \\"style\\",
      \\"architect\\": {},
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    },
    \\"test-ui-form\\": {
      \\"projectType\\": \\"library\\",
      \\"root\\": \\"libs/test/ui-form\\",
      \\"sourceRoot\\": \\"libs/test/ui-form/src\\",
      \\"prefix\\": \\"form\\",
      \\"architect\\": {
        \\"lint\\": {
          \\"builder\\": \\"@nrwl/linter:eslint\\",
          \\"options\\": {
            \\"lintFilePatterns\\": [\\"libs/test/ui-form/src/**/*.ts\\"]
          }
        },
        \\"test\\": {
          \\"builder\\": \\"@nrwl/jest:jest\\",
          \\"options\\": {
            \\"jestConfig\\": \\"libs/test/ui-form/jest.config.js\\",
            \\"passWithNoTests\\": true
          }
        }
      },
      \\"schematics\\": {
        \\"@schematics/angular:component\\": {
          \\"style\\": \\"scss\\"
        }
      }
    }
  },
  \\"newProjectRoot\\": \\"\\",
  \\"cli\\": {
    \\"defaultCollection\\": \\"@nrwl/nest\\"
  },
  \\"defaultProject\\": \\"api\\",
  \\"schematics\\": {
    \\"@nrwl/angular:application\\": {
      \\"unitTestRunner\\": \\"jest\\",
      \\"e2eTestRunner\\": \\"cypress\\"
    },
    \\"@nrwl/angular:library\\": {
      \\"unitTestRunner\\": \\"jest\\"
    }
  }
}
"
`;

exports[`init schematic should run successfully 3`] = `
"{
  \\"compilerOptions\\": {
    \\"paths\\": {
      \\"@nxpm/api-app-module\\": [\\"apps/api/src/app/app.module.ts\\"],
      \\"@proj/api/auth/data-access\\": [\\"libs/api/auth/data-access/src/index.ts\\"],
      \\"@proj/api/auth/feature\\": [\\"libs/api/auth/feature/src/index.ts\\"],
      \\"@proj/api/core/data-access\\": [\\"libs/api/core/data-access/src/index.ts\\"],
      \\"@proj/api/core/feature\\": [\\"libs/api/core/feature/src/index.ts\\"],
      \\"@proj/test/assets\\": [\\"libs/test/assets/src/index.ts\\"],
      \\"@proj/test/auth/data-access\\": [\\"libs/test/auth/data-access/src/index.ts\\"],
      \\"@proj/test/auth/ui\\": [\\"libs/test/auth/ui/src/index.ts\\"],
      \\"@proj/test/core/data-access\\": [\\"libs/test/core/data-access/src/index.ts\\"],
      \\"@proj/test/about/feature\\": [\\"libs/test/about/feature/src/index.ts\\"],
      \\"@proj/test/auth/feature\\": [\\"libs/test/auth/feature/src/index.ts\\"],
      \\"@proj/test/core/feature\\": [\\"libs/test/core/feature/src/index.ts\\"],
      \\"@proj/test/dashboard/feature\\": [\\"libs/test/dashboard/feature/src/index.ts\\"],
      \\"@proj/test/shell/feature\\": [\\"libs/test/shell/feature/src/index.ts\\"],
      \\"@proj/test/layout\\": [\\"libs/test/layout/src/index.ts\\"],
      \\"@proj/test/style\\": [\\"libs/test/style/src/index.ts\\"],
      \\"@proj/test/ui-form\\": [\\"libs/test/ui-form/src/index.ts\\"]
    }
  }
}
"
`;

exports[`init schematic should run successfully 4`] = `
"{
  \\"name\\": \\"test-name\\",
  \\"dependencies\\": {
    \\"@angular/animations\\": \\"^10.1.0\\",
    \\"@angular/common\\": \\"^10.1.0\\",
    \\"@angular/compiler\\": \\"^10.1.0\\",
    \\"@angular/core\\": \\"^10.1.0\\",
    \\"@angular/forms\\": \\"^10.1.0\\",
    \\"@angular/platform-browser\\": \\"^10.1.0\\",
    \\"@angular/platform-browser-dynamic\\": \\"^10.1.0\\",
    \\"@angular/router\\": \\"^10.1.0\\",
    \\"@apollo/client\\": \\"^3.2.9\\",
    \\"@kikstart-playground/graphql-intercom\\": \\"1.4.1\\",
    \\"@nestjs/common\\": \\"^7.0.0\\",
    \\"@nestjs/config\\": \\"^0.5.0\\",
    \\"@nestjs/core\\": \\"^7.0.0\\",
    \\"@nestjs/graphql\\": \\"^7.7.0\\",
    \\"@nestjs/jwt\\": \\"^7.1.0\\",
    \\"@nestjs/passport\\": \\"^7.1.0\\",
    \\"@nestjs/platform-express\\": \\"^7.0.0\\",
    \\"@ngrx/component-store\\": \\"10.1.2\\",
    \\"@ngx-formly/bootstrap\\": \\"^5.10.8\\",
    \\"@ngx-formly/core\\": \\"^5.10.8\\",
    \\"@prisma/cli\\": \\"^2.13.0\\",
    \\"@prisma/client\\": \\"^2.13.0\\",
    \\"apollo-angular\\": \\"^2.1.0\\",
    \\"apollo-server-express\\": \\"^2.18.0\\",
    \\"bcryptjs\\": \\"^2.4.3\\",
    \\"bootstrap\\": \\"^4.5.3\\",
    \\"bootswatch\\": \\"^4.5.3\\",
    \\"class-transformer\\": \\"^0.3.1\\",
    \\"class-validator\\": \\"^0.12.2\\",
    \\"cookie-parser\\": \\"1.4.5\\",
    \\"graphql\\": \\"^15.3.0\\",
    \\"graphql-tools\\": \\"^6.2.3\\",
    \\"graphql-type-json\\": \\"^0.3.2\\",
    \\"joi\\": \\"^17.2.1\\",
    \\"passport\\": \\"^0.4.1\\",
    \\"passport-jwt\\": \\"^4.0.0\\",
    \\"reflect-metadata\\": \\"^0.1.13\\",
    \\"rxjs\\": \\"~6.5.5\\",
    \\"supertest\\": \\"^5.0.0\\",
    \\"tslib\\": \\"^2.0.0\\",
    \\"zone.js\\": \\"^0.10.2\\"
  },
  \\"devDependencies\\": {
    \\"@angular-devkit/build-angular\\": \\"~0.1001.3\\",
    \\"@angular/compiler-cli\\": \\"^10.1.0\\",
    \\"@angular/language-service\\": \\"^10.1.0\\",
    \\"@graphql-codegen/cli\\": \\"1.19.3\\",
    \\"@graphql-codegen/introspection\\": \\"1.18.1\\",
    \\"@graphql-codegen/typescript\\": \\"1.18.1\\",
    \\"@graphql-codegen/typescript-apollo-angular\\": \\"^2.2.1\\",
    \\"@graphql-codegen/typescript-operations\\": \\"1.17.12\\",
    \\"@nestjs/schematics\\": \\"^7.0.0\\",
    \\"@nestjs/testing\\": \\"^7.0.0\\",
    \\"@nrwl/cypress\\": \\"10.4.4\\",
    \\"@nrwl/eslint-plugin-nx\\": \\"10.4.4\\",
    \\"@nrwl/jest\\": \\"10.4.4\\",
    \\"@nrwl/nest\\": \\"10.4.4\\",
    \\"@nrwl/node\\": \\"10.4.4\\",
    \\"@nxpm/cli\\": \\"^4.1.2\\",
    \\"@types/jest\\": \\"26.0.8\\",
    \\"@typescript-eslint/eslint-plugin\\": \\"4.3.0\\",
    \\"@typescript-eslint/parser\\": \\"4.3.0\\",
    \\"codelyzer\\": \\"~5.0.1\\",
    \\"cypress\\": \\"^5.5.0\\",
    \\"eslint\\": \\"7.10.0\\",
    \\"eslint-config-prettier\\": \\"6.0.0\\",
    \\"eslint-plugin-cypress\\": \\"^2.10.3\\",
    \\"husky\\": \\"^4.3.0\\",
    \\"jest\\": \\"26.2.2\\",
    \\"jest-preset-angular\\": \\"8.3.1\\",
    \\"lint-staged\\": \\"^10.4.0\\",
    \\"ts-jest\\": \\"26.4.0\\",
    \\"typescript\\": \\"~4.0.2\\"
  },
  \\"husky\\": {
    \\"hooks\\": {
      \\"pre-commit\\": \\"lint-staged\\",
      \\"pre-push\\": \\"yarn format:check\\"
    }
  },
  \\"lint-staged\\": {
    \\"nx.json\\": [
      \\"yarn nxpm-stack lint\\",
      \\"yarn format:write\\"
    ],
    \\"*.{js,json,css,scss,md,ts,html,graphql}\\": [
      \\"yarn format --uncommitted\\"
    ]
  },
  \\"scripts\\": {
    \\"dev:test\\": \\"nx serve test --hmr\\",
    \\"build:test\\": \\"nx build test --prod\\",
    \\"sdk:test\\": \\"graphql-codegen --config libs/test/core/data-access/src/codegen.yml\\",
    \\"sdk:test:watch\\": \\"yarn sdk:test --watch\\",
    \\"dev:api\\": \\"nx serve api\\",
    \\"build:api\\": \\"nx build api --prod\\",
    \\"setup\\": \\"yarn prisma:apply\\",
    \\"prisma:apply\\": \\"yarn prisma:format && yarn prisma db push --preview-feature\\",
    \\"prisma:format\\": \\"yarn prisma format\\",
    \\"prisma:generate\\": \\"yarn prisma generate\\",
    \\"prisma:migrate\\": \\"yarn prisma migrate save --experimental && yarn prisma migrate up --experimental\\",
    \\"prisma:studio\\": \\"yarn prisma studio\\",
    \\"build\\": \\"yarn build:test && yarn build:api\\",
    \\"start\\": \\"node dist/apps/api/main.js\\",
    \\"test:ci\\": \\"yarn prisma:apply && yarn e2e api-e2e\\",
    \\"docker:build\\": \\"docker build . -t nxpm/api\\",
    \\"docker:run\\": \\"docker run -it -p 8000:3000 nxpm/api\\",
    \\"docker:push\\": \\"docker push nxpm/api\\",
    \\"postinstall\\": \\"ngcc --properties es2015 browser module main --first-only --create-ivy-entry-points\\",
    \\"e2e\\": \\"ng e2e\\"
  },
  \\"prisma\\": {
    \\"schema\\": \\"libs/api/core/data-access/src/prisma/schema.prisma\\"
  }
}
"
`;
